include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -309,6 +312,54 @@ MultigridPreconditionerBase<dim, Number>::check_levels(std::vector<MGLevelInfo>
   }
 }
 
+#if DEAL_II_VERSION_GTE(10, 0, 0)
+
+/**
+ * A class to use for the deal.II coarsening functionality, where we try to
+ * balance the mesh coarsening with a minimum granularity and the number of
+ * partitions on coarser levels.
+ */
+template<int dim, int spacedim = dim>
+class BalancedGranularityPartitionPolicy : public RepartitioningPolicyTools::Base<dim, spacedim>
+{
+public:
+  BalancedGranularityPartitionPolicy(types::global_cell_index const n_cells_fine,
+                                     unsigned int const             n_mpi_processes)
+    : n_cells_per_level{n_cells_fine}, n_mpi_processes_per_level{n_mpi_processes}
+  {
+  }
+
+  virtual LinearAlgebra::distributed::Vector<double>
+  partition(Triangulation<dim, spacedim> const & tria_coarse_in) const override
+  {
+    types::global_cell_index const n_cells = tria_coarse_in.n_global_active_cells();
+
+    // TODO: We hard-code a grain-size limit of 200 cells per processor
+    // (assuming linear finite elements and typical behavior of
+    // supercomputers), but we also do not want to coarsen by more than a
+    // factor of 8 to not create too many MPI messages.
```

Comment:
I don't understand the comment "coarsen by more than a factor of 8" completely (does it refer to the mesh or the MPI processes?). I would read the code like reducing the number of MPI processes by a factor of 8 for the next coarser level (at least as a target value). Maybe also information is missing here whether the vectors are filled from fine-to-coarse or coarse-to-fine.

Reply:
Updated comment in 4e6b0ff.